mountpoint-s3/tests/fuse_tests/mod.rs

```diff
@@ -263,3 +287,22 @@ pub fn read_dir_to_entry_names(read_dir_iter: ReadDir) -> Vec<String> {
         })
         .collect::<Vec<_>>()
 }
+
+#[macro_export]
+macro_rules! sleep_till_retry_succeed {
+    ($method_result:expr) => {
+        for i in 1..5 {
+            match $method_result {
+                Ok(bool_value) => {
+                    if bool_value {
+                        break;
+                    } else {
+                        sleep(Duration::from_millis(100 * i));
+                    }
+                }
+                Err(_) => panic!("The provided method could not provide result"),
```

Comment:
I'd like to see us be able to provide the message here on what we're waiting for